CMakeLists.txt

```diff
@@ -100,40 +103,26 @@ if (NOT CMAKE_BUILD_TYPE)
     set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "debug" "release")
 endif()
 
-# When we find threads, prefer not to use the -pthread option
-# in order to avoid a CMake 3.12 issue.
-set(THREADS_PREFER_PTHREAD_FLAG OFF)
-
-# Add CUDA as a language if GPU support requested.
-# (This has to be set early so as to enable CUDA tests in generator
-# expressions.)
-
+# Add CUDA as a language if GPU support requested. (This has to be set early so
+# as to enable CUDA tests in generator expressions.)
 if(ARB_GPU STREQUAL "cuda")
     set(ARB_WITH_NVCC TRUE)
-
-    # CMake 18 and later set the default CUDA architecture for
-    # each target according to CMAKE_CUDA_ARCHITECTURES.
-    if (NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
-        set(CMAKE_CUDA_ARCHITECTURES 60 70 80)
-    endif()
-
-    # This fixes nvcc picking up a wrong host compiler for linking, causing issues
-    # with outdated libraries, eg libstdc++ and std::filesystem. Must happend before
-    # all calls to enable_language(CUDA)
+    # This fixes nvcc picking up a wrong host compiler for linking, causing
+    # issues with outdated libraries, eg libstdc++ and std::filesystem. Must
+    # happen before all calls to enable_language(CUDA)
     set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
-
     enable_language(CUDA)
-
-    # Despite native CUDA support, the CUDA package is still required to export
-    # the cuda library dependencies from the installed target.
-    find_package(CUDA 10 REQUIRED)
-
+    find_package(CUDAToolkit)
+    if(NOT DEFINED CMAKE_CUDA_ARCHITECTURES)
+        if(CUDAToolkit_VERSION_MAJOR GREATER 10)
+            set(CMAKE_CUDA_ARCHITECTURES 60 70 80)
+        else()
+            set(CMAKE_CUDA_ARCHITECTURES 60 70)
+        endif()
+    endif()
 elseif(ARB_GPU STREQUAL "cuda-clang")
     set(ARB_WITH_CUDA_CLANG TRUE)
-
-    # The CUDA package is needed for clang compilation for the same reasons as above.
-    # enable_langaue(CUDA) has a bug with clang
-    find_package(CUDA 10 REQUIRED)
+    enable_language(CUDA)
```

Comment:
Duplicate of line 114.

Reply:
Yes, since we are here checking for CUDA-clang and CUDA in l114